js/source/tile.js

```diff
@@ -129,11 +129,11 @@ class Tile {
     }
 
     redoPlacement(source) {
-        if (source.type !== 'vector' && source.type !== 'geojson') {
+        if ((source.type !== 'vector' && source.type !== 'geojson') ||
+            !this.hasData() || !this.collisionTile) {
```

Comment:
Does this affect the behavior if `redoPlacement` is called before the tile is initially loaded? For example, if the map is rotated while tiles are loading, it may be necessary to redo placement immediately after the load completes.